### PR TITLE
Adding requset count agg

### PR DESCRIPTION
**Pull request by shilmyhasan**
- Added a new aggregation named APIM_ReqCountAgg in APIM_THROTTLED_OUT_SUMMARY.siddhi 
 file
- adding regionalID in abnormal request count Alert
- changes in template with the changes in alert (adding regionalID)

### CODE REVIEW

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi [hunk 1]
```diff
@@ -39,11 +39,31 @@ define stream ThrottledOutStream(
...
 
+@store(type='rdbms', datasource='APIM_ANALYTICS_DB')
+@purge(enable='true', interval='60 min', @retentionPeriod(sec='1 day', min='72 hours', hours='90 days', days='1 year', months='2 years', years='3 years'))
+define aggregation APIM_ReqCountAgg
+from reqCountTempStream
+select apiName, apiVersion, apiCreator, apiCreatorTenantDomain, applicationName, sum(successCount) as successCount, sum(throttleCount) as throttleCount
+group by apiName, apiVersion
+aggregate by timestamp every seconds...year;
+
+from ThrottledOutStream left outer join Request
```
Comment (dilini-muthumala) [inline features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi]: We are joining two streams here. Did this work?

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi [hunk 2]
```diff
@@ -39,11 +39,31 @@ define stream ThrottledOutStream(
...
+select ThrottledOutStream.apiName as apiName, ThrottledOutStream.apiVersion as apiVersion, ThrottledOutStream.apiCreator as apiCreator, ThrottledOutStream.apiCreatorTenantDomain as apiCreatorTenantDomain, ThrottledOutStream.applicationName as applicationName, 0 as successCount, 1 as throttleCount, ThrottledOutStream.throttledOutTimestamp as timestamp
+insert into reqCountTempStream;
+
+from Request left outer join ThrottledOutStream
```
Comment (dilini-muthumala) [inline features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi]: Same for here also. We are joining two streams. Have we tested this scenario?

Comment (shilmyhasan) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi]: Yes, we had to join the two streams and form a Temporary Stream in order to create an aggregation which receives events from both Throttle and Request Streams

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi (hunk 1 above)
Comment (shilmyhasan) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_THROTTLED_OUT_SUMMARY.siddhi]: yes, It works.

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_ABNORMAL_REQUEST_COUNT.siddhi [hunk 3]
```diff
@@ -44,22 +44,22 @@ define table ApimAllAlert (type string, tenantDomain string, message string, sev
 -- This aggregation definition is only for retrieving data. No data is actually aggregated from this.
 @store(type = 'rdbms', datasource = 'APIM_ANALYTICS_DB')
 define aggregation ApiUserPerAppAgg
-from Request 
-select apiName, apiVersion, apiContext, apiCreator, apiCreatorTenantDomain, apiHostname, applicationId, applicationName, applicationOwner, username, userTenantDomain, count(apiName) as totalRequestCount, gatewayType, label 
+from Request
+select apiName, apiVersion, apiContext, apiCreator, apiCreatorTenantDomain, apiHostname, applicationId, applicationName, applicationOwner, username, userTenantDomain, count() as totalRequestCount, gatewayType, label,'default' as regionalID
```
Comment (praminda) [inline features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_ABNORMAL_REQUEST_COUNT.siddhi]:
```diff
--,'default' as regionalID
++ ,'default' as regionalID

Comment (shilmyhasan) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/APIM_ABNORMAL_REQUEST_COUNT.siddhi]: Fixed in 705b19ecd44e0d78cc4f30342c6d4c9c9d7c4337

**Pull request merged**